static/js/invite.js

```diff
@@ -87,9 +87,16 @@ exports.initialize = function () {
                               .show();
             } else {
                 // Some users were not invited.
+                var invitee_emails_errored = [];
+                var invitee_emails_left = arr.sent_invitations ? [] : invitee_emails.val().split(/\n|,/);
                 var error_list = $('<ul>');
                 _.each(arr.errors, function (value) {
                     error_list.append($('<li>').text(value.join(': ')));
+                    invitee_emails_errored.push(value[0]);
+                    var index = invitee_emails_left.indexOf(value[0]);
+                    if (index >= 0) {
+                        invitee_emails_left.splice(index, 1);
+                    }
```

Comment:
The rearranging function might be also useful in other contexts, so I agree with Tim that extracting it makes sense.